.helm/ecamp3/templates/api_deployment.yaml

```diff
@@ -3,18 +3,19 @@ kind: Deployment
 metadata:
   name: {{ include "api.name" . }}
   labels:
-    {{- include "app.labels" . | nindent 4 }}
+    {{- include "api.selectorLabels" . | nindent 4 }}
+    {{- include "app.commonLabels" . | nindent 4 }}
 spec:
   {{- if not .Values.autoscaling.enabled }}
   replicas: {{ .Values.api.replicaCount }}
   {{- end }}
   selector:
     matchLabels:
-      {{- include "app.selectorLabels" . | nindent 6 }}
+      {{- include "api.selectorLabels" . | nindent 6 }}
```

Comment:
**More info on why this change is necessary:**

before this would output the following labels (same labels for each "deployment" on dev):
```yaml
matchLabels:
      app.kubernetes.io/name: ecamp3
      app.kubernetes.io/instance: ecamp3-dev
      app.kubernetes.io/part-of: ecamp3
```

now this will output:
```yaml
matchLabels:
      app.kubernetes.io/name: ecamp3-api
      app.kubernetes.io/instance: ecamp3-dev
      app.kubernetes.io/part-of: ecamp3
```

See documentation on https://kubernetes.io/docs/concepts/workloads/controllers/deployment/ for why this change is important and why label selectors need to be unique for each Deployment:

> Do not overlap labels or selectors with other controllers (including other Deployments and StatefulSets). Kubernetes doesn't stop you from overlapping, and if multiple controllers have overlapping selectors those controllers might conflict and behave unexpectedly.

This change will probably not be upgradeable directly. Needs a manual `helm uninstall` before we can deploy again.
https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#label-selector-updates

> In API version apps/v1, a Deployment's label selector is immutable after it gets created.